### PR TITLE
[WIP] ui: setup basic multi-browser tests with Playwright

### PR DESCRIPTION
**Pull request by terrcin**
Currently we don't do any browser testing with playwright, but rather only use playwright to get auth credentials to use in some direct API calls. Our current setup doesn't allow for multi-browser testing either which we need to have.

This PR moves to using the [Playwright Test runner](https://playwright.dev/docs/test-intro) which is designed for running multi-browser tests with. This means moving to a different test file structure and separating out Playwright configuration into a dedicated file. Initially there are two simple tests to show that the user can login and navigate the site and see the correct content that are run against Chromium, Firefox, and Webkit

The old placeholder API calling UI tests are there and run still

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -58,6 +58,10 @@ packages/buildinfo/src/buildinfo.ts
 
 *.jwt
 
+# playwright output
+test/test-remote/uishot-*
```
Comment (MoSattler) [inline .gitignore:62]: Would it maybe make sense to have a screenshot directory in `test-remote` that can be ignored?

**Comment by MoSattler** [comment]
@terrcin this is somewhat related to #806, let's make sure we're not stepping on each other's toes!

File: .gitignore (hunk 1 above)
Comment (terrcin) [reply .gitignore:62]: Agree. Although, I'm trying to keep my structure changes to a minimium with this PR though so have not changed how screenshots work or are saved. I put this in so I didn't accidently commit them.

**Comment by terrcin** [comment]
@MoSattler our toes should be safe here. :-) This PR leaves the existing Playwright tests there and running so any newly added tests to `test-remote/test_ui.ts` will be fine. Once this PR merges it'll be real easy to move accross any newly added browser tests to be under the Playwright Test runner code path.

**Comment by terrcin** [comment]
Progress! got past all the linting failures and now the tests have actually tried to run, but failed.
https://buildkite.com/opstrace/prs/builds/4608#370a9175-cc82-48f4-89ef-256da2e85008/10733-10734

```
[2021-06-13T00:29:26Z] 	yarn playwright test
--
&nbsp; | yarn run v1.22.5
&nbsp; | warning Skipping preferred cache folder "/var/lib/buildkite-agent/.cache/yarn" because it is not writable.
&nbsp; | warning Selected the next writable cache folder in the list, will be "/tmp/.yarn-cache-2000".
&nbsp; | $ playwright test
&nbsp; | warning Cannot find a suitable global folder. Tried these: "/usr/local, /var/lib/buildkite-agent/.yarn"
&nbsp; | [2021-06-13T00:29:27Z] Using config at /build/test/test-remote/playwright.config.ts
&nbsp; | [2021-06-13T00:29:27Z] (node:29) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
&nbsp; | [2021-06-13T00:29:27Z] (Use `node --trace-deprecation ...` to show where the warning was created)
&nbsp; | [2021-06-13T00:29:27Z]
&nbsp; | [2021-06-13T00:29:27Z] Running 12 tests using 2 workers
&nbsp; | [2021-06-13T00:29:27Z]
&nbsp; | x 8) tests/authentication.spec.ts:25:3 › [WebKit] after auth0 authentication user should see own email in user list (0ms)[1E
&nbsp; | [2021-06-13T12:33:00Z] TypeError: Cannot read property 'config' of undefinedeprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
&nbsp; | [2021-06-13T12:33:00Z]   pw:api &lt;= browserType.launch failed +9msw where the warning was created)
&nbsp; | [2021-06-13T12:33:00Z]     at WorkerRunner._deadline (/build/node_modules/@playwright/test/lib/test/workerRunner.js:73:30)
&nbsp; | [2021-06-13T12:33:00Z]     at WorkerRunner.cleanup (/build/node_modules/@playwright/test/lib/test/workerRunner.js:53:20)
&nbsp; | [2021-06-13T12:33:00Z]     at gracefullyCloseAndExit (/build/node_modules/@playwright/test/lib/test/worker.js:107:32)
&nbsp; | [2021-06-13T12:33:00Z]     at process.&lt;anonymous&gt; (/build/node_modules/@playwright/test/lib/test/worker.js:89:15)
&nbsp; | [2021-06-13T12:33:00Z]     at process.emit (events.js:376:20)
&nbsp; | [2021-06-13T12:33:00Z]     at emit (internal/child_process.js:910:12)
&nbsp; | [2021-06-13T12:33:00Z]     at processTicksAndRejections (internal/process/task_queues.js:83:21)
&nbsp; | x 9) tests/authentication.spec.ts:25:3 › [Firefox] after auth0 authentication user should see own email in user list (0ms)[1E
&nbsp; | [2021-06-13T12:33:00Z] TypeError: Cannot read property 'config' of undefined
&nbsp; | [2021-06-13T12:33:00Z]
&nbsp; | [2021-06-13T12:33:00Z]     at WorkerRunner._deadline (/build/node_modules/@playwright/test/lib/test/workerRunner.js:73:30)
&nbsp; | [2021-06-13T12:33:00Z]     at WorkerRunner.cleanup (/build/node_modules/@playwright/test/lib/test/workerRunner.js:53:20)
&nbsp; | [2021-06-13T12:33:00Z]     at gracefullyCloseAndExit (/build/node_modules/@playwright/test/lib/test/worker.js:107:32)
&nbsp; | [2021-06-13T12:33:00Z]     at process.&lt;anonymous&gt; (/build/node_modules/@playwright/test/lib/test/worker.js:89:15)
&nbsp; | [2021-06-13T12:33:00Z]     at process.emit (events.js:376:20)
&nbsp; | [2021-06-13T12:33:00Z]     at emit (internal/child_process.js:910:12)
&nbsp; | [2021-06-13T12:33:00Z]     at processTicksAndRejections (internal/process/task_queues.js:83:21)
&nbsp; | [2021-06-13T12:33:00Z]
&nbsp; | [2021-06-13T12:33:00Z]
&nbsp; | [2021-06-13T12:33:00Z] Error: ENOENT: no such file or directory, lstat '/build/test/test-remote/tests/authentication.spec.ts'
&nbsp; | error Command failed with exit code 1.
&nbsp; | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
&nbsp; | [2021-06-13T12:33:00Z] make[1]: *** [Makefile:669: test-remote-browser] Error 1
&nbsp; | [2021-06-13T12:33:00Z] make[1]: Leaving directory '/build'
&nbsp; | [2021-06-13T12:33:00Z] + EXITCODE_MAKE_TESTREMOTE_BROWSER=2
&nbsp; | [2021-06-13T12:33:00Z] + set -e
&nbsp; | [2021-06-13T12:33:00Z] + echo '--- Exit status of make test-remote-browser: 2'
```

**Comment by terrcin** [comment]
another ci failure: https://buildkite.com/opstrace/prs/builds/4609#5340fd9a-1508-4035-88d5-a7daf56f2866/11157-11194

```
[2021-06-13T21:48:06Z] 	npx playwright test --project=Firefox
--
&nbsp; | [2021-06-13T21:48:07Z] Using config at /build/test/test-remote/playwright.config.ts
&nbsp; | [2021-06-13T21:48:07Z] (node:1) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
&nbsp; | [2021-06-13T21:48:07Z] (Use `node --trace-deprecation ...` to show where the warning was created)
&nbsp; | [2021-06-13T21:48:07Z]
&nbsp; | [2021-06-13T21:48:07Z] Running 4 tests using 2 workers
&nbsp; | [2021-06-13T21:48:07Z]
&nbsp; | x 1) tests/authentication.spec.ts:21:3 › [Firefox] after auth0 authentication user should see homepage (11ms)[3E    tests/authentication.spec.ts:25:3 › [Firefox] after auth0 authentication user should see own email in user list:
&nbsp; | [2021-06-13T21:48:08Z] (node:25) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
&nbsp; | [2021-06-13T21:48:08Z] (Use `node --trace-deprecation ...` to show where the warning was created)
&nbsp; | [2021-06-13T21:48:08Z]   pw:api =&gt; browserType.launch started +0ms
&nbsp; | [2021-06-13T21:48:08Z]   pw:api &lt;= browserType.launch failed +5ms
```


going to try more debug info on the next run

**Comment by terrcin** [comment]
CI stalled again. I think the solution is going to me instead of re-using the image we already had for running tests (I was doing that to try and be quick about it) but to make one following the official instructions for `playwright test` - there might even be a base image we can use. These multi-browser tests just need to run a headless browser and know what url is needed and the hardcoded ci user creds. nothing else nor fancy so it should be fairly straight forward.

**Comment by jgehrcke** [comment]
I have manually canceled both hanging builds (aws/gcp) in https://buildkite.com/opstrace/prs/builds/4611#300845c8-bd60-4eda-a511-84d71c3fcaeb to free up the agent -- the cloud resources need to be manually cleaned up now.

**Comment by terrcin** [comment]
closing as has been superseded by https://github.com/opstrace/opstrace/pull/895

**Pull request closed without merging**